src/components/button/Button.tsx

```diff
@@ -5,128 +5,214 @@
         <div>
-          <div className={styles.icon}>
-            <span className={styles["icon-left"]}>{this.props.iconLeft}</span>
-            <span className={styles.label}>{this.props.children}</span>
-          </div>
-          <Loader loading={this.props.loading} percentage={this.props.percentageDone}/>
+            <Label>{this.props.children}</Label >
+            <Loader loading={this.props.loading} percentage={this.props.percentageDone}/>
         </div>
       )
-
     }
 
-    if (hasIcon && this.props.iconRight) {
+    if (hasIcon && this.props.iconLeft) {
       buttonContent = (
         <div>
-          <div className={styles.icon}>
-            <span className={styles.label}>{this.props.children}</span>
-            <span className={styles["icon-right"]}>{this.props.iconRight}</span>
-          </div>
+           <Icon>
+            <IconLeft>{this.props.iconLeft}</IconLeft>
+            <Label>{this.props.children}</Label>
+          </Icon>
           <Loader loading={this.props.loading} percentage={this.props.percentageDone}/>
         </div>
       )
     }
 
-    if (hasIcon && this.props.iconOnly) {
+    if (hasIcon && this.props.iconRight) {
       buttonContent = (
         <div>
-          <div className={styles.icon}>
-            {this.props.iconOnly}
-          </div>
+          <Icon>
+            <Label>{this.props.children}</Label>
+            <IconRight>{this.props.iconRight}</IconRight>
+          </Icon>
+          <Loader loading={this.props.loading} percentage={this.props.percentageDone}/>
         </div>
       )
     }
 
-    if (!hasIcon) {
+    if (hasIcon && this.props.iconOnly) {
       buttonContent = (
         <div>
-          <span className={styles.label}>{this.props.children}</span>
-          <Loader loading={this.props.loading} percentage={this.props.percentageDone}/>
+          <IconOnly>{this.props.iconOnly}</IconOnly>
         </div>
       )
     }
 
-
-    const className = cx(
-      'button',
-      {[`button--${this.props.color}`]: !!this.props.color},
-      {'button--wide': this.props.wide},
-      {'button--spaced': this.props.spaced},
-      {'button--clear': this.props.clear},
-      {'button--small': this.props.size === 'small'},
-      {'button--large': this.props.size === 'large'},
-      {'button--loading': this.props.loading},
-      {'button--outlined': this.props.outlined},
-      {'button--round': this.props.rounded},
-      {'button--icon': hasIcon},
-      this.props.classNames,
-    );
-
     return (
       //@ts-ignore
-      <Tag
+      <NormalButton
         id={this.props.id}
-        className={className}
-        onClick={this.props.onClick}
         disabled={isDisabled}
-        href={this.props.href}
+        color={this.props.color}
+        wide={this.props.wide}
         type={this.props.type}
-        target={this.props.target}>
+        size={this.props.size}
+        clear={this.props.clear}
+        //clicker moet hier komen, werkt niet meer
```

Comment:
Met clicker bedoel je het click event? Zou wel moeten werken, da's het belangrijkste van een knop :) 

Nog een tip: als er nog todo's zijn die je in code wil commenten, prefix ze dan met "Todo: " op die manier kunnen we daar op zoeken.

Reply:
Er was een error op, ik ging zometeen verder kijken om dit op te lossen zodat hij terug werkt. :-)